src/main/scala/com/socrata/querycoordinator/Service.scala

```diff
@@ -468,9 +475,24 @@ class Service(secondaryProvider: ServiceProviderProvider[AuxiliaryData],
     def loop() {
       in.read(buf) match {
         case -1 => // done
-        case n => out.write(buf, 0, n); loop()
+        case n =>
+          out.write(buf, 0, n);
+          loop()
       }
     }
     loop()
   }
-}
+
+  private def transferHeaders(resp: HttpResponse, headers: Map[String, Seq[String]]): HttpResponse = {
+    headers.foldLeft(resp) { case (acc, (h: String, vs: Seq[String])) =>
+      vs.foldLeft(acc) { (acc2, v: String) =>
+        acc2 ~> Header(h, v)
+      }
+    }
+  }
+
+  private def resetResponse(response: HttpServletResponse): Unit = {
+    if (response.isCommitted) ???
```

Comment:
Not implemented is not new.  It is carried over from  if(resp.isCommitted) line 436.